### ISSUE TITLE
AutoCompletion nor Treesitter working

### ISSUE THREAD

**Issue by Nizarll** [lopi-py/luau-lsp.nvim#3]
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/151186173/919406fa-ee45-4e3b-8f8e-b1326ee4675a)

Here's my mason_lspconfig
plugin is installed and everything is okay idk why its not working ?

btw im new on nvim

**Comment by lopi-py** [comment]
* `sourcemap` and `types` options should not be there, please read the README.md
* `require("luau-lsp").treesitter()` should be *before* your tree-sitter config (again, check the README.md), after that you need to run `:TSInstall luau`
* make sure that the lsp is actually installed via mason

**Comment by Nizarll** [comment]
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/151186173/64aace1f-5578-493e-808f-07041efd20df)
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/151186173/8e60ed43-803f-4e3a-ae82-015cffb8704b)

**Comment by Nizarll** [comment]
Still not working also i got this error when putting require('luau-lsp').treesitter() before the tree sitter configuration file 
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/151186173/76cb03b5-907b-49f6-b569-1a5c95a72220)

**Comment by lopi-py** [comment]
for treesitter to work you need to reinstall the psrser. whats your output of `:LspInfo` when you enter a luau buffer? (luau lsp work for projects, not standalone scripts)

**Comment by Nizarll** [comment]
So I can't see if the Lsp is working on a simple luau script ?

**Comment by lopi-py** [comment]
Nope, you need to work in a luau project with root markers

**Comment by jasonpanosso** [comment]
I can't tell if the treesitter parser isn't working, or if the way `vim.lsp.buf.hover` renders luau is janky.

This is from a hover request on nvim nightly:
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/36612827/8f730ba5-9b17-4fa9-9ff0-f60d16d0479c)

Here is a hover request from nvim v0.9.4/v0.9.1(same for both):
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/36612827/2a72b199-af11-4a28-b35d-8f4b9ea34ef4)

Strangely, it actually looks (mostly?) fine within a markdown file:
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/36612827/0943a955-abbe-4586-8e44-b50c5ebe8329)

I've verified that I ran `require("luau-lsp").treesitter()` before my treesitter config, and my `lua-lsp.nvim` config is the same as the example. I've also verified that I uninstalled the original luau parser, and reinstalled after adding `require("luau-lsp").treesitter()` to my config.

I'm using nvim-treesitter commit 3da5cfb

Not sure if my issue is related, however I didn't want to open a new issue as they seemed similar. Would appreciate any help :)

**Comment by lopi-py** [comment]
Yeah its working as expected, only on nightly nvim makes markdown stuff to use treesitter (thats why only hover renders properly on nightly 🙂)

**Comment by jasonpanosso** [comment]
Any clue why it would highlight/color the function keyword differently between a markdown file and hover? the `function DataModel:GetService(className: string): Instance` example looks different between the markdown file and the hover, and I have no idea why :P

**Comment by lopi-py** [comment]
That is not related to this, it depends on how luau lsp returns the hover markdown, If you notice, function signatures aren't actual luau, just a pseudo-like luau so its expected to not render properly, vscode has also this problem afaik

**Comment by lopi-py** [comment]
That is not related to this, it depends on how luau lsp returns the hover markdown, If you notice, function signatures aren't actual luau, just a pseudo-like luau so its expected to not render properly, vscode has also this problem afaik

**Comment by lopi-py** [comment]
@Nizarll that happens when you have the psrser running so you need to not run the luau parser (dont enter any buffer when you are reinstalling) or delete the parser manually (you have got the path in the screenshot I believe)

**Comment by lopi-py** [comment]
I'll document better the stuff about the parser and make it optional in case you want to use the default one

**Comment by jasonpanosso** [comment]
> That is not related to this, it depends on how luau lsp returns the hover markdown, If you notice, function signatures aren't actual luau, just a pseudo-like luau so its expected to not render properly, vscode has also this problem afaik, but it is better than just a plain and uncolored text!

That makes sense, I'm assuming that pre v.10/nightly used `markdown_fenced_languages` to render markdown, which is why there was no highlighting as it simply didn't recognize luau.

However, I'm still confused why these get highlighted differently:

hover - 
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/36612827/ab783067-ee18-4df5-8141-fcf5fce5e119)

markdown file - 
![image](https://github.com/lopi-py/luau-lsp.nvim/assets/36612827/f5a20d8e-6f17-4d18-ae3a-4178026da972)

Those appear to be the exact same, and I'm assuming that a markdown file and hover both use treesitter on nightly, so I'm wondering why they would behave different o_O. Doesn't seem to be an issue with this plugin, but do you experience the same issue by chance?

**Comment by jasonpanosso** [comment]
I would have assumed that treesitter would be used in both markdown files, and hover, as if you run `:set filetype?` in the context of hover, and the context of a markdown file both return that the filetype is `markdown`. 

Also the markdown file is the exact same as the hover, I just copy pasted that single line from the hover into it.

Super disturbing behavior, but luau is super jank so I'm not really surprised. I'm not rly bothered about it, I just wanted to make sure that it isn't just me seeing this behavior

**Comment by lopi-py** [comment]
Yeah I would asume it has to do with the parser but idk
